packages/rancher-alerting-drivers/charts/templates/hardened.yaml

```diff
@@ -0,0 +1,117 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "drivers.fullname" . }}-patch-sa
+  namespace: {{ .Release.Namespace }}
+  labels: {{ include "drivers.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": post-install, post-upgrade
```

Comment:
Should this be pre-install/upgrade, as the default SA already exists. That way it gets edited before the custom SA gets installed.

Reply:
The edit on the default SA can happen at either pre or post of the install/upgrade. 
I can move it to pre-install/upgrade if that makes the logic more clear.

Reply:
It's post-install since Monitoring V2 needed this job to run after the Namespace resource was created by Helm. It's not a requirement for Alerting Drivers tho, since Alerting Drivers doesn't create namespaced Helm resources.

We could change it, but IMO we should just keep it the same if possible.

Reply:
I'm ok with post, on one stipulation. I don't want to run into a race with hardened cluster where the default SA is attempted to be used before it gets patched away.

Reply:
The namespace is created either by Rancher UI or the user prior to installing the chart, so we do not know if the default SA is used by others or not.  What we can confirm is that the rancher-alerting-driver app does not use the default SA. 

As to the case where the default SA is attempted to be used during the period between starting installing the rancher-alerting-drivers app and executing the post hook, how is it different from the default SA is attempted to be used at any time prior to installing the app? IMO, we do not need to worry about it, or that is something the system, not this app, needs to handle.

Reply:
Oh, I was thinking about this wrong. The default SA is only applied if the app doesn't specify a service account. As long as it does the default SA won't be used.